dace/transformation/dataflow/map_distribution.py

```diff
@@ -0,0 +1,940 @@
+                        for idx in new_access.min_element():
+                            if not isinstance(idx, Number):
+                                return False
+                            if idx != 0:
+                                stencil_found = True
+                    else:
+                        first_access = a
+                    indices = a.min_element()
+                    unmatched_indices = set(params)
+                    for idx in indices:
+                        if isinstance(idx, sympy.Symbol):
+                            bidx = idx
+                        elif isinstance(idx, sympy.Add):
+                            if len(idx.free_symbols) != 1:
+                                return False
+                            bidx = list(idx.free_symbols)[0]
+                        else:
+                            return False
+                        if bidx in unmatched_indices:
+                            unmatched_indices.remove(bidx)
+                    if len(unmatched_indices) > 0:
+                        return False
+            else:
+                return False
+
+        outputs = dict()
+        for _, _, _, _, m in graph.in_edges(map_exit):
+            if m.wcr:
+                return False
+            desc = sdfg.arrays[m.data]
+            if desc not in outputs.keys():
+                outputs[desc] = []
+            outputs[desc].append(m.subset)
+        
+        for desc, accesses in outputs.items():
+            if isinstance(desc, (dace.data.Array, dace.data.View)):
+                for a in accesses:
+                    if a.num_elements() > 1:
+                        return False
+                    indices = a.min_element()
+                    unmatched_indices = set(params)
+                    for idx in indices:
+                        if isinstance(idx, sympy.Symbol):
+                            bidx = idx
+                        elif isinstance(idx, sympy.Add):
+                            if len(idx.free_symbols) != 1:
+                                return False
+                            bidx = list(idx.free_symbols)[0]
+                        else:
+                            return False
+                        if bidx in unmatched_indices:
+                            unmatched_indices.remove(bidx)
+                    if len(unmatched_indices) > 0:
+                        return False
+            else:
+                return False
+
+        return stencil_found
+
+    @staticmethod
+    def match_to_str(graph: dace.SDFGState, candidate: Dict[pm.PatternNode,
+                                                            int]) -> str:
+        map_entry = graph.node(candidate[StencilOperation.map_entry])
+        return map_entry.map.label + ': ' + str(map_entry.map.params)
+
+    def apply(self, sdfg: dace.SDFG):
+        pass
+
+
+@registry.autoregister_params(singlestate=True)
+class OuterProductOperation(pm.Transformation):
```

Comment:
Same as above. I'd rather have those in separate files/PRs if we don't use them now.